migrations/Version202404141320101555_taoAdvancedSearch.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\Exception\IrreversibleMigration;
+use oat\oatbox\reporting\Report;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\taoAdvancedSearch\scripts\install\RegisterTaskQueueServices;
+use oat\taoTaskQueue\model\Service\QueueAssociationService;
+use oat\taoTaskQueue\scripts\tools\BrokerFactory;
+
+final class Version202404141320101555_taoAdvancedSearch extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Register missing queue associations.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $registrationService = new RegisterTaskQueueServices();
+        $this->propagate($registrationService);
+        $registrationService->__invoke([]);
+
+        if (BrokerFactory::BROKER_MEMORY !== $this->getAssocitationService()->guessDefaultBrokerType()) {
```

Comment:
We are repeating the same migration content as https://github.com/oat-sa/extension-tao-advanced-search/blob/develop/migrations/Version202106011320101488_taoAdvancedSearch.php#L25

Maybe it worth to encapsulate this instruction in `RegisterTaskQueueServices` including this `if` validation so we avoid discrepancy? Wdyt?

Reply:
it does different things. I agree that it is almost the same migration, but it what I can move out is like 2-3 lines of code, or remove the warning below which is not necessary to have.